### PR TITLE
VAPE NATION take 2

### PR DESCRIPTION
**Pull request by tortellinitony**
https://github.com/tgstation/tgstation/pull/20629

Adds vape and a gnarly shirt to show off that it's just vape bro!

### CODE REVIEW

**Comment by dreamcrab** [comment]
does it blow smoke clouds that people cant see through

**Comment by tortellinitony** [comment]
if emagged it makes big smoke clouds of whatever chemical you have

**Comment by Nebyab** [comment]
Isn't this an in-progress pr? 

I have mild concerns about porting it, if that is the case.

Otherwise, sure why not.

**Comment by tortellinitony** [comment]
TG merged it and it works as far as I can tell, but whatever you think.

**Comment by doobiedoo23** [comment]
> he didn't name his branch vapestation

+1

**Comment by KayeArray** [comment]
YES

**Comment by ghost** [comment]
GIVE US ANOTHER ONE
GIVE US ANOTHER ONE
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+1 even tho it's pretty degenerate ;^)

**Comment by Cuboos** [comment]
Darn hippies stealing my PR.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+			param_color = pick("red","blue","black","white","green","purple","yellow","orange")
+		icon_state = "[param_color]_vape"
+		item_state = "[param_color]_vape"
+
+/obj/item/clothing/mask/vape/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/weapon/reagent_containers/glass))
+		if(reagents.total_volume < chem_volume)
+			if(O.reagents.total_volume > 0)
+				O.reagents.trans_to(src,25)
+				user << "<span class='notice'>You add the contents of [O] to the [src]</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a . to the end.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 2]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+			else
+				user << "<span class='warning'>The [O] is empty!</span>"
+		else
+			user << "<span class='warning'>[src] can't hold anymore reagents!</span>"
+
+	if(istype(O, /obj/item/weapon/screwdriver))
+		if(!screw)
+			screw = 1
+			user << "<span class='notice'>You open the cap on the [src]</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a . to the end.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 3]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+			if(super)
+				var/image/I = (image(icon, "vapeopen_med"))
+				overlays += I
+			else
+				var/image/I = (image(icon, "vapeopen_low"))
+				overlays += I
+		else
+			screw = 0
+			user << "<span class='notice'>You close the cap on the [src]</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a . to the end.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 4]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+			overlays.Cut()
+
+	if(istype(O, /obj/item/device/multitool))
+		if(screw && !emagged)//also kinky
+			if(!super)
+				overlays.Cut()
+				super = 1
+				user << "<span class='notice'>You increase the voltage in the [src]</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a . to the end.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 5]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+				var/image/I = (image(icon, "vapeopen_med"))
+				overlays += I
+			else
+				overlays.Cut()
+				super = 0
+				user << "<span class='notice'>You decrease the voltage in the [src]</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a . to the end.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 6]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+			user << "<span class='notice'>The [name] can't be modified!</span>"
+
+
+/obj/item/clothing/mask/vape/emag_act(mob/user)// I WON'T REGRET WRITTING THIS, SURLY.
+	if(screw)
+		if(!emagged)
+			overlays.Cut()
+			emagged = 1
+			super = 0
+			user << "<span class='warning'>You maximize the voltage in the [src]</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a ! to the end.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 7]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+/obj/item/clothing/mask/vape/attack_self(mob/user)
+	if(reagents.total_volume > 0)
+		user << "<span class='notice'>you empty [src] of all reagents.</span>"
+		reagents.clear_reagents()
+	return
+
+/obj/item/clothing/mask/vape/equipped(mob/user, slot)
+	if(slot == slot_wear_mask)
+		if(!screw)
+			user << "<span class='notice'>You start puffing on that dank vape</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a . to the end.

File: code/modules/clothing/suits/miscellaneous.dm [hunk 8]
```diff
@@ -272,6 +272,12 @@
 	icon_state = "nerdshirt"
 	item_state = "nerdshirt"
 
+/obj/item/clothing/suit/vapeshirt //wearing this is asking to get beat.
+	name = "Vape Naysh shirt"
```
Comment (ghost) [inline code/modules/clothing/suits/miscellaneous.dm]: Aren't names meant not to start with capitals?

File: code/modules/clothing/suits/miscellaneous.dm [hunk 9]
```diff
@@ -272,6 +272,12 @@
...
+	desc = "A cheap white T-shirt with a big tacky \"VN\" on the front, Why would you wear this unironicly?"
```
Comment (ghost) [inline code/modules/clothing/suits/miscellaneous.dm]: Unironically.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 10]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+			user << "<span class='warning'>You maximize the voltage in the [src]</span>"
+			var/image/I = (image(icon, "vapeopen_high"))
+			overlays += I
+			var/datum/effect_system/spark_spread/sp = new /datum/effect_system/spark_spread //for effect
+			sp.set_up(5, 1, src)
+			sp.start()
+		else
+			user << "<span class='warning'>The [name] is already emagged!</span>"
+	else
+		user << "<span class='notice'>You need to open the cap to do that</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Add a . to the end.

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 11]
```diff
@@ -617,3 +617,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+
+/obj/item/clothing/mask/vape/attack_self(mob/user)
+	if(reagents.total_volume > 0)
+		user << "<span class='notice'>you empty [src] of all reagents.</span>"
```
Comment (ghost) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Capital letter required.

**Review by ghost — CHANGES_REQUESTED** [review]
Fix up the text please.

**Comment by ghost** [comment]
Hold on that +1 for me until you fix up the flavor text please.

**Comment by ArcLumin** [comment]
@Cuboos You know it lad.

File: code/modules/clothing/suits/miscellaneous.dm (hunk 9 above)
Comment (tortellinitony) [reply code/modules/clothing/suits/miscellaneous.dm]: How culd they murge such a atrosity

**Comment by tortellinitony** [comment]
Fixed the grammar

**Comment by SplergeBucket** [comment]
hel yeha

**Comment by tortellinitony** [comment]
By the way, you can vape donuts...

**Comment by FuryMcFlurry** [comment]
not a bug to me

**Comment by Pyko1** [comment]
fix the missing dot that mcbaws wanted to be fixed and then it should be ready to be merged.

**Comment by tortellinitony** [comment]
the punctuation should be all fixed now

**Comment by Pyko1** [comment]
seems fine, merging.

**Comment by KayeArray** [comment]
yes boys

**Comment by optimumtact** [comment]
it's tg not TG

**Pull request merged**